### ISSUE TITLE
Nested timelines weird preview

### ISSUE THREAD

**Issue by Animmaniac** [Scirra/Construct-bugs#7178]
## Problem description

When using nested timelines, keyframed properties have an effect outside the time boundaries of the nested timelines, giving unexpected results when previewing in editor.

Also nested timelines don't seem to inherit their properties like 'Result mode' from their parent timeline.

## Attach a .c3p

[nested-timelines-playback-bug-A04.zip](https://github.com/Scirra/Construct-bugs/files/11977203/nested-timelines-playback-bug-A04.zip)

## Steps to reproduce

1. Open each of the timelines named 'move', 'rotate', and 'scale' to see that they each modify only one property.
2.  Observe the Sprite in the layout. Enter editing mode for 'Timeline 1'. Observe the Sprite again at time 0. Then play 'Timeline 1' in the editor.
3. Play 'Timeline 1' again and observe the Sprite in the layout.
4. Scrub the time mark all the way to the end, then quickly all the way to the start. Observe the angle of the Sprite in the layout. Repeat if necessary.
5. Preview the project and observe the result.

## Observed result

In step 2, the Sprite rotation and scale start different in editing mode than in the editor, even thought at the beginning of 'Timeline 1' only the position is being modified through the nested timeline 'move'. It seems that frame 0 of the 'rotation' and 'scale' timelines "leak" before they even start.

Editor | Editing mode
![nested-timelines-playback-bug-A02](https://github.com/Scirra/Construct-bugs/assets/29487913/8ab27e4f-5795-47a8-8bb2-4624693b4950)

In step 3, notice that the Sprite now starts rotated 45º and slimmer than previously, and uses the last state of the last play as a starting point, instead of starting fresh.
![nested-timelines-playback-bug-C01](https://github.com/Scirra/Construct-bugs/assets/29487913/4e4ed494-28a9-4e98-93e8-268e8d15f492)

In step 4, if you happen to scrub quickly enough, the Sprite can end up having an intermediate rotation (and scale) when you move the time mark to frame 0 of 'Timeline 1', even thought it should be perfectly horizontal. It seems that scrubbing only keeps the last rendered angle state within the boundaries of the 'rotate' timeline, instead of displaying the current state of the time mark (the same for other nested timelines out of bound and their properties).
![nested-timelines-playback-bug-D01](https://github.com/Scirra/Construct-bugs/assets/29487913/e98b29bd-dca8-4678-91ba-7031640d6437)

In step 5, the project preview doesn't show the problems that are apparent in the editor, and at first glance it seems to be alright. However 'Timeline 1' 'Result mode' is set to 'Absolute', and all it's nested timelines have 'Result mode' set to 'Default', so it's expected they should inherit 'Absolute' from 'Timeline 1' while nested. But at the end of the preview, the Sprite will be much thicker than compared to the absolute result of the 'scale' timeline, and with different rotation, as if it was applying relative transformations.
![nested-timelines-playback-bug-E01](https://github.com/Scirra/Construct-bugs/assets/29487913/bcb9d897-0ddb-4241-b449-98e08a34ed48)

## Expected result

In step 2, the Sprite editing state should be identical to the layout state before. Only it's position should be modified at the beginning, and the 'rotation' and 'scale' timelines should not leak outside their time boundaries.
![nested-timelines-playback-bug-B02](https://github.com/Scirra/Construct-bugs/assets/29487913/97d02224-5e0a-429a-b7e7-466adce1eb81)

In step 3, the new play state should start equal to last time, and not be dependent to the last play state.
![nested-timelines-playback-bug-B02](https://github.com/Scirra/Construct-bugs/assets/29487913/97d02224-5e0a-429a-b7e7-466adce1eb81)

In step 4, the Sprite should display it's state at time 0, ignoring properties that are not keyframed on 'Timeline 1' yet, and not display intermediate rotations from the last scrubbing.
![nested-timelines-playback-bug-B02](https://github.com/Scirra/Construct-bugs/assets/29487913/97d02224-5e0a-429a-b7e7-466adce1eb81)

In step 5, the nested timelines should inherit the 'Absolute' 'Result mode' from 'Timeline 1', and the displayed Sprite position, angle and size at the end of preview should be identical as if each nested timeline had 'Result mode' set to 'Absolute'.
![nested-timelines-playback-bug-F01](https://github.com/Scirra/Construct-bugs/assets/29487913/89623ced-9cd2-486d-b290-451e877c6276)

## More details

**Affected browsers/platforms:**  Tested on Chrome 114.0.5735.199

**First affected release:**  Tested on Construct Animate r348 (beta)

## System details

<details><summary>View details</summary>

Platform information
Product: Construct Animate r348 (beta)
Browser: Chrome 114.0.5735.199
Browser engine: Chromium
Context: browser
Operating system: Windows 10
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
Language setting: en-US

Local storage
Storage quota (approx): 224 gb
Storage usage (approx): 159 mb (0.1%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Supports highp precision: yes
Vendor: Google Inc. (Intel)
Renderer: ANGLE (Intel, Intel(R) HD Graphics 4600 Direct3D11 vs_5_0 ps_5_0, D3D11)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_draw_buffers_indexed
OES_texture_float_linear
OVR_multiview2
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
WEBGL_provoking_vertex
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.265 (video/mp4; codecs=hev1.1.2.L93.B0)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM AV1 (video/webm; codecs=av1)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by DiegoScirra** [comment]
The issues regarding, incorrect preview and nested timelines not getting the result mode and animation mode from parent timelines will be fixed in the next beta.

I did find that the runtime was not producing the results I would have expected from using 'Relative' result mode. That has been changed too. Now, if you play 'Timeline 1' by itself on 'Relative' or 'Absolute' modes, the results look the same. The difference will only be noticeable if there are other actions trying to affect the instance, in that case 'Absolute' mode makes it so the timeline overrides everything.
